File: src/Entity/Matches.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;
use Symfony\Component\Routing\Annotation\Route;

/**
 * Matches
 *
 * @ORM\Table(name="matches", indexes={@ORM\Index(name="matches_fk1", columns={"id_Team_2"}), @ORM\Index(name="matches_fk0", columns={"id_Team_1"})})
 * @ORM\Entity
 */
class Matches
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_match", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     * @Groups("matches")
     */
    private $idMatch;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="time", type="time", nullable=false)
     * @Groups("matches")
     */
    private $time;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date", type="date", nullable=false)
     * @Groups("matches")
     */
    private $date;

    /**
     * @var string
     *
     * @ORM\Column(name="location", type="string", length=255, nullable=false)
     * @Assert\Length(
     *      min = 4,
     *      max = 20,
     *      minMessage = "Your location must be at least {{ limit }} characters long",
     *      maxMessage = "Your location cannot be longer than {{ limit }} characters"
     * )
     * @Groups("matches")
     */
    private $location;

    /**
     * @var int|null
     *
     * @ORM\Column(name="nb_seats", type="integer", nullable=true)
     * @Assert\NotBlank
     * @Groups("matches")
     */
    private $nbSeats;

    /**
     * @var int
     *
     * @ORM\Column(name="price", type="integer", nullable=false)
     * @Assert\NotBlank
     * @Groups("matches")
     */
    private $price;

    /**
     * @var string|null
     *
     * @ORM\Column(name="link", type="string", length=255, nullable=true)
     * @Assert\Length(
     *      min = 4,
     *      max = 20,
     *      minMessage = "Your link must be at least {{ limit }} characters long",
     *      maxMessage = "Your link cannot be longer than {{ limit }} characters"
     * )
     * @Groups("matches")

     */
    private $link;

    /**
     * @var \Team
     *
     * @ORM\ManyToOne(targetEntity="Team")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_Team_1", referencedColumnName="id_team")
     * })
     * @Groups("matches")
     */
    private $idTeam1;

    /**
     * @var \Team
     *
     * @ORM\ManyToOne(targetEntity="Team")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_Team_2", referencedColumnName="id_team")
     * })
     * @Groups("matches")
     */
    private $idTeam2;

    public function getIdMatch(): ?int
    {
        return $this->idMatch;
    }

    public function getTime(): ?\DateTimeInterface
    {
        return $this->time;
    }

    public function setTime(\DateTimeInterface $time): self
    {
        $this->time = $time;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getLocation(): ?string
    {
        return $this->location;
    }

    public function setLocation(string $location): self
    {
        $this->location = $location;

        return $this;
    }

    public function getNbSeats(): ?int
    {
        return $this->nbSeats;
    }

    public function setNbSeats(?int $nbSeats): self
    {
        $this->nbSeats = $nbSeats;

        return $this;
    }

    public function getPrice(): ?int
    {
        return $this->price;
    }

    public function setPrice(int $price): self
    {
        $this->price = $price;

        return $this;
    }

    public function getLink(): ?string
    {
        return $this->link;
    }

    public function setLink(?string $link): self
    {
        $this->link = $link;

        return $this;
    }

    public function getIdTeam1(): ?Team
    {
        return $this->idTeam1;
    }

    public function setIdTeam1(?Team $idTeam1): self
    {
        $this->idTeam1 = $idTeam1;

        return $this;
    }

    public function getIdTeam2(): ?Team
    {
        return $this->idTeam2;
    }

    public function setIdTeam2(?Team $idTeam2): self
    {
        $this->idTeam2 = $idTeam2;

        return $this;
    }




}

File: src/com/mycompany/entites/Matches.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.entites;


/**
 *
 * @author Ghassene Badra
 */
public class Matches {
    
    private int idMatch;
    private String time;
    private String date;
    private String location,link;
    private int nbSeats,price;
    private int idTeam1,idTeam2;
    private String teamName1, teamName2;

    public String getTeamName1() {
        return teamName1;
    }

    public void setTeamName1(String teamName1) {
        this.teamName1 = teamName1;
    }

    public String getTeamName2() {
        return teamName2;
    }

    public void setTeamName2(String teamName2) {
        this.teamName2 = teamName2;
    }

    public Matches(int idMatch, String time, String date, String location, String link, int nbSeats, int price, int idTeam1, int idTeam2) {
        this.idMatch = idMatch;
        this.time = time;
        this.date = date;
        this.location = location;
        this.link = link;
        this.nbSeats = nbSeats;
        this.price = price;
        this.idTeam1 = idTeam1;
        this.idTeam2 = idTeam2;
    }

    public Matches(String time, String date, String location, String link, int nbSeats, int price, int idTeam1, int idTeam2) {
        this.time = time;
        this.date = date;
        this.location = location;
        this.link = link;
        this.nbSeats = nbSeats;
        this.price = price;
        this.idTeam1 = idTeam1;
        this.idTeam2 = idTeam2;
    }

    public Matches() {
    }

    public Matches(String time, String date, String location, String link) {
        this.time = time;
        this.date = date;
        this.location = location;
        this.link = link;
    }

    public Matches(String location, String link, int nbSeats) {
        this.location = location;
        this.link = link;
        this.nbSeats = nbSeats;
    }

    public Matches(String time, String date, String location, String link, int nbSeats, int price) {
        this.time = time;
        this.date = date;
        this.location = location;
        this.link = link;
        this.nbSeats = nbSeats;
        this.price = price;
    }

    public Matches(String location, String link, int nbSeats, int price) {
        this.location = location;
        this.link = link;
        this.nbSeats = nbSeats;
        this.price = price;
    }

    public Matches(String location, String link, int nbSeats, int price, int idTeam1, int idTeam2) {
        this.location = location;
        this.link = link;
        this.nbSeats = nbSeats;
        this.price = price;
        this.idTeam1 = idTeam1;
        this.idTeam2 = idTeam2;
    }
    
    
    
    

    public int getIdMatch() {
        return idMatch;
    }

    public void setIdMatch(int idMatch) {
        this.idMatch = idMatch;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public int getNbSeats() {
        return nbSeats;
    }

    public void setNbSeats(int nbSeats) {
        this.nbSeats = nbSeats;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public int getIdTeam1() {
        return idTeam1;
    }

    public void setIdTeam1(int idTeam1) {
        this.idTeam1 = idTeam1;
    }

    public int getIdTeam2() {
        return idTeam2;
    }

    public void setIdTeam2(int idTeam2) {
        this.idTeam2 = idTeam2;
    }

    @Override
    public String toString() {
        return "Matches{" + "idMatch=" + idMatch + ", time=" + time + ", date=" + date + ", location=" + location + ", link=" + link + ", nbSeats=" + nbSeats + ", price=" + price + ", idTeam1=" + idTeam1 + ", idTeam2=" + idTeam2 + ", teamName1=" + teamName1 + ", teamName2=" + teamName2 + '}';
    }


    
    
    
     
}
